neuro_extras/image.py

```diff
@@ -380,6 +383,36 @@ async def _build_image(
             raise click.ClickException(f"Failed to build image: {exit_code}")
 
 
+async def _select_build_preset(client) -> Optional[str]:
+    """
+    Try to automatically select the best available preset for image build.
+    """
+    # Require minimum 2 vCPU
+    min_cpu = 2
+    # Require minimum 4 GB RAM
+    min_mem = 4096
+    good_presets = []
+    # Build a shortlist of presets that
+    for cluster_preset_name, cluster_preset_info in client.presets.items():
+        # Don't even try to use GPU machines for image builds
+        if (
+            cluster_preset_info.cpu >= min_cpu
+            and cluster_preset_info.memory_mb >= min_mem
+            and cluster_preset_info.gpu is None
```

Comment:
Maybe, we could relax this requirement and still add the preset (for instance, if it's preemptible).
Later we in any pick the cheapest one.